### ISSUE TITLE
Auto detect when SDK not able to run on a device

### ISSUE THREAD

**Issue by bfouet-gpsw** [googlesamples/cardboard-java#51]
Hi,

I've written an app to play a bit with the Cardboard SDK, and I've tried to get it to run on the devices I have at hand.
One of them is a samsung galaxy tab running Android 4.1.2 (fingerprint: samsung/espresso10wifixx/espresso10wifi:4.1.2/JZO54K/P5110XXDMC2:user/release-keys), advertising gles 2.0 with the following version / extensions:
GL_VENDOR: Imagination Technologies
GL_RENDERER: PowerVR SGX 540
GL_VERSION: OpenGL ES 2.0 build 1.9.RC2@2139099
GL_EXTENSIONS: GL_EXT_debug_marker
 GL_OES_rgb8_rgba8
 GL_OES_depth24
 GL_OES_vertex_half_float
 GL_OES_texture_float
 GL_OES_texture_half_float
 GL_OES_element_index_uint
 GL_OES_mapbuffer
 GL_OES_fragment_precision_high
 GL_OES_compressed_ETC1_RGB8_texture
 GL_OES_EGL_image
 GL_OES_EGL_image_external
 GL_OES_required_internalformat
 GL_OES_depth_texture
 GL_OES_get_program_binary
 GL_OES_packed_depth_stencil
 GL_OES_standard_derivatives
 GL_OES_vertex_array_object
 GL_OES_egl_sync
 GL_EXT_multi_draw_arrays
 GL_EXT_texture_format_BGRA8888
 GL_EXT_discard_framebuffer
 GL_EXT_shader_texture_lod
 GL_IMG_shader_binary
 GL_IMG_texture_compression_pvrtc
 GL_IMG_texture_npot
 GL_IMG_texture_format_BGRA8888
 GL_IMG_read_format
 GL_IMG_program_binary
 GL_IMG_uniform_buffer_object
 GL_IMG_multisampled_render_to_texture

I have the following traces output in logcat:
D/CardboardViewNativeImpl: Native library loaded
I/dalvikvm: Could not find method android.opengl.EGL14.eglSwapInterval, referenced from method com.google.vrtoolkit.cardboard.CardboardViewNativeImpl$RendererHelper.onDrawFrame
W/dalvikvm: VFY: unable to resolve static method 1027: Landroid/opengl/EGL14;.eglSwapInterval (Landroid/opengl/EGLDisplay;I)Z
D/dalvikvm: VFY: replacing opcode 0x71 at 0x0053

Same thing for eglGetCurrentSurface.

Are there plans to detect automatically issues like this? If already done, how can an SDK user can be informed, in order to provide some feedback to their own users?

Thanks,
Ben

**Comment by nathanmartz** [comment]
Hi, Ben. We don't have any plans to support auto-detection like this, as it would just tell users what they already know - the app doesn't work on their device. The correct approach is to use your app manifest (we supply a sample one in the Cardboard SDK) as well as the whitelist/blacklist features in the Play store to ensure you only run on compatible devices.

**Comment by bfouet-gpsw** [comment]
Are you able to provide some sort of a list of the supported devices, based on testing on your side, and what we users have reported?
At least, having a way to get the error from the SDK instead of the silent fail would help to build the black list dynamically: when a device fails, an error could be sent back to the developer to help him build the list.

Please note that I get your point, I'm just trying to find a way to ease our lives as developers on top of the SDK.

**Comment by dav-cz** [comment]
Hi, just curious, do you happen to have a rooted phone?
Our code doesn't use eglSwapInterval on phone with Android api 4.1.

Could you report what Build.VERSION.SDK_INT returns on your phone?

**Comment by bfouet-gpsw** [comment]
Hi @dav-cz, the returned version is 16.
I just got side-tracked by the messages in the logcat actually. I think the real issue on this tablet is that is has no sensor support so installing my app on it makes little sense anyway.
I just tried modifying my renderer a bit so that it forces matrix it receives to the identity, and I can then see the display works just fine.
(a lot of noise for nothing, sorry about that :-) )
